adam_preprocessing/shape_stroke_graph_inference.py

```diff
@@ -106,32 +112,45 @@ def main():
     best_acc = 0
 
     "Inference"
-    outputs = nn.LogSoftmax(dim=1)(
-        model(test_adjacency_matrices, test_node_features, test_edge_features)
-    )
-    test_acc = Variable(
-        evaluation(
-            outputs.data,
-            test_label.data,
-            topk=(1,),
-        )[0]
-    )
-    logging.info("test acc :{}".format(test_acc))
+    if len(test_coords) > 0:
+        outputs = nn.LogSoftmax(dim=1)(
+            model(test_adjacency_matrices, test_node_features, test_edge_features)
+        )
+    else:
+        # if we know there's only one input, we know there should only be one output, so we CAN give
+        # a placeholder for now
+        if args.dir_num is not None:
+            logging.warning(
+                "Can't give reasonable output for empty input. Outputting default decode of 0 (%s)",
+                STRING_OBJECT_LABELS[0],
+            )
+            outputs = torch.zeros((1, 1))
```

Comment:
If we have no strokes maybe the reasonable solution is to just not provide any output and call it a miss from processing error? Then we'd just have output with no language from ADAM but that's "ok" if scene processing has failed

Reply:
That works for me, and I think that works fine for the demo, but I am slightly worried about a crash in batch mode. I don't recall how well ADAM handles it when it's missing a feature file/the feature file is missing any actual features.

Reply:
My guess is that it'll just wreck the pattern but I'm not sure.

Reply:
Based on `phase3_load_from_disk()` it looks like when loading we should end up with no feature files when we glob for them, so `len(feature_yamls) != 1`, so we will try to load `action.yaml` which doesn't exist and will crash.

Reply:
Ah, ok, I'd want to update this so we still provide a `feature.yaml` file just the stroke features would be empty. That way we import an empty file and ADAM doesn't crash.

Reply:
I think that makes sense.

Reply:
I think you want to drop the None check on [line 138](https://github.com/isi-vista/adam/pull/1159/files#diff-66e64562fb1b22bb6e3daf6ac5d35aa750717246813016ea80e206d8cb193ce8R138) so that we produce blank/placeholder output when there are no strokes to use to identify the input. But note this only produces placeholder output within the context of the demo.

We have to extend the code if we want to do this for a general curriculum, i.e. not just within the demo. That would probably involve changes to the stroke extraction code too. That shouldn't be hard but it's involved enough I think it's not worth tackling in this PR.

Reply:
Agreed with handling the more complex interaction in another PR. I've relocated the check for `outputs` being None to guard against the assert clause for now.

Reply:
Sounds good.